src/modules/jcms_rest/src/Plugin/rest/resource/PromotionalCollectionItemRestResource.php

```diff
@@ -0,0 +1,157 @@
+ *   label = @Translation("Promotional collection item rest resource"),
+ *   uri_paths = {
+ *     "canonical" = "/promotional-collections/{id}"
+ *   }
+ * )
+ */
+class PromotionalCollectionItemRestResource extends AbstractRestResourceBase {
+
+  /**
+   * Responds to GET requests.
+   *
+   * Returns a list of bundles for specified entity.
+   *
+   * @throws JCMSNotFoundHttpException
+   */
+  public function get(string $id) : JCMSRestResponse {
+    if ($this->checkId($id)) {
+      $query = \Drupal::entityQuery('node')
+        ->condition('changed', \Drupal::time()->getRequestTime(), '<')
+        ->condition('type', 'promotional_collection')
+        ->condition('uuid', '%' . $id, 'LIKE');
+
+      if (!$this->viewUnpublished()) {
+        $query->condition('status', NodeInterface::PUBLISHED);
+      }
+
+      $nids = $query->execute();
+      if ($nids) {
+        $nid = reset($nids);
+        /* @var \Drupal\node\Entity\Node $node */
+        $node = Node::load($nid);
+
+        $this->setSortBy('changed');
+        $response = $this->processDefault($node, $id);
+
+        // Image is optional.
+        if ($image = $this->processFieldImage($node->get('field_image'), FALSE)) {
+          $attribution = $this->fieldValueFormatted($node->get('field_image_attribution'), FALSE, TRUE);
+          if (!empty($attribution)) {
+            foreach ($image as $key => $type) {
+              $image[$key]['attribution'] = $attribution;
+            }
+          }
+          $response['image'] = $image;
+        }
+
+        // Impact statement is optional.
+        if ($node->get('field_impact_statement')->count()) {
+          $response['impactStatement'] = $this->fieldValueFormatted($node->get('field_impact_statement'));
+          if (empty($response['impactStatement'])) {
+            unset($response['impactStatement']);
+          }
+        }
+
+        // Subjects are optional.
+        $subjects = $this->processSubjects($node->get('field_subjects'));
+        if (!empty($subjects)) {
+          $response['subjects'] = $subjects;
+        }
+
+        // Editors are optional.
+        if ($node->get('field_editors')->count()) {
+          $people_rest_resource = new PersonListRestResource([], 'person_list_rest_resource', [], $this->serializerFormats, $this->logger);
+          $response['editors'] = [];
+          foreach ($node->get('field_editors')->referencedEntities() as $editor) {
+            /* @var Node $editor */
+            if ($editor->isPublished() || $this->viewUnpublished()) {
+              $editor_item = $people_rest_resource->getItem($editor);
+              $response['editors'][] = $editor_item;
+            }
+          }
+        }
+
+        // Summary is optional.
+        if ($content = $this->processFieldContent($node->get('field_summary'))) {
+          $response['summary'] = $content;
+        }
+
+        // Promotional collection content is required.
+        $response['content'] = [];
```

Comment:
I think it is reasonable to remove some duplicate code here that can be shared with `CollectionItemRestResource`

Reply:
I will refactor this resource after this is merged in: https://github.com/elifesciences/journal-cms/pull/508